### DISCUSSION TITLE
Lesson 7 Part 21 test testUserCanFundInteractions FAIL

### DISCUSSION THREAD

**Discussion by DuncanDuMond** [Cyfrin/foundry-full-course-cu#388] [Q&A] [answered]
I ran forge test --match-test testUserCanFundInteractions in the terminal and received the following error: 

[⠰] Compiling...
[⠔] Compiling 1 files with 0.8.19
[⠑] Solc 0.8.19 finished in 1.09s
Compiler run successful!

Running 1 test for test/integration/InteractionsTest.t.sol:InteractionsTest
[FAIL. Reason: FundMe__NotOwner()] testUserCanFundInteractions() (gas: 644620)
Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 907.25µs
Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/integration/InteractionsTest.t.sol:InteractionsTest
[FAIL. Reason: FundMe__NotOwner()] testUserCanFundInteractions() (gas: 644620)

Encountered a total of 1 failing tests, 0 tests succeeded

I think it is due to the video skipping at the end of this segment where the code I was supposed to use is missing but do not know where. Here is my code from the InteractionsTest.t.sol file:

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Test, console} from "forge-std/Test.sol";
import {FundMe} from "../../src/FundMe.sol";
import {DeployFundMe} from "../../script/DeployFundMe.s.sol";
import {FundFundMe, WithdrawFundMe} from "../../script/Interactions.s.sol";

contract InteractionsTest is Test {
    FundMe fundMe;

    address USER = makeAddr("user");
    uint256 constant SEND_VALUE = 0.1 ether; // 100000000000000000
    uint256 constant STARTING_BALANCE = 10 ether; // 10000000000000000000
    uint256 constant GAS_PRICE = 1; // 10000000000 = 10 gwei

    function setUp() external {
        DeployFundMe deployFundMe = new DeployFundMe();
        fundMe = deployFundMe.run();
        vm.deal(USER, STARTING_BALANCE);
    }

    function testUserCanFundInteractions() public {
        FundFundMe fundFundMe = new FundFundMe();
        fundFundMe.fundFundMe(address(fundMe));

        WithdrawFundMe withdrawFundMe = new WithdrawFundMe();
        withdrawFundMe.withdrawFundMe(address(fundMe));

        assert(address(fundMe).balance == 0);
    }
}

I am also putting my code for the Interactions.s.sol file for reference only.

// SPDX-License-Identifier: MIT

pragma solidity ^0.8.19;

import {Script, console} from "forge-std/Script.sol";
import {DevOpsTools} from "foundry-devops/src/DevOpsTools.sol";
import {FundMe} from "../src/FundMe.sol";

contract FundFundMe is Script {
    uint256 constant SEND_VALUE = 0.01 ether; // 100000000000000000

    function fundFundMe(address mostRecentlyDeployed) public {
        vm.startBroadcast();
        FundMe(payable(mostRecentlyDeployed)).fund{value: SEND_VALUE}();
        vm.stopBroadcast();
        console.log("Funded FundMe with %s", SEND_VALUE);
    }

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
            "FundMe",
            block.chainid
        );
        vm.startBroadcast();
        fundFundMe(mostRecentlyDeployed);
        vm.stopBroadcast();
    }
}

contract WithdrawFundMe is Script {
    function withdrawFundMe(address mostRecentlyDeployed) public {
        FundMe(payable(mostRecentlyDeployed)).withdraw();
    }

    function run() external {
        address mostRecentlyDeployed = DevOpsTools.get_most_recent_deployment(
            "FundMe",
            block.chainid
        );
        vm.startBroadcast();
        withdrawFundMe(mostRecentlyDeployed);
        vm.stopBroadcast();
    }
}

Please advise.

**Comment by pyromaus** [comment, marked as answer]
in `WithdrawFundMe`, you have this line:
```solidity
FundMe(payable(mostRecentlyDeployed)).withdraw();
```
Add this line above it:
```solidity
vm.broadcast();
FundMe(payable(mostRecentlyDeployed)).withdraw();
```
Without `broadcast`, you're trying to call `withdraw()` as the script, not as the `owner`. Hence the error `FundMe__NotOwner()`
Clean up your question formatting so we can help you better

**Comment by DuncanDuMond** [comment]
Ok I got it to work. Thanks!
